test.js

```diff
@@ -1,147 +1,107 @@
-const addButton = document.getElementById('add-button');
-const errorMessage = document.getElementById('error-message');
-
-// Task storage (in-memory for now)
-let tasks = [];
-
-// Function to render tasks to the screen
-function renderTasks() {
-  taskList.innerHTML = ''; // Clear the list before rendering
-
-  if (tasks.length === 0) {
-    taskList.innerHTML = '<li>No tasks to show</li>';
-  } else {
-    tasks.forEach(task => {
-      const taskElement = document.createElement('li');
-      taskElement.className = task.completed ? 'completed' : '';
-
-      const taskText = document.createElement('span');
-      taskText.textContent = task.text;
-      taskText.className = 'task-text';
-      taskElement.appendChild(taskText);
-
-      // Edit Button
-      const editButton = document.createElement('button');
-      editButton.textContent = 'Edit';
-      editButton.className = 'edit-btn';
-      editButton.onclick = () => editTask(task.id);
-      taskElement.appendChild(editButton);
-
-      // Delete Button (Forgot to add a delete button handler)
-      const deleteButton = document.createElement('button');
-      deleteButton.textContent = 'Delete';
-      deleteButton.className = 'delete-btn';
-      taskElement.appendChild(deleteButton);
-
-      // Mark as Completed Button
-      const completeButton = document.createElement('button');
-      completeButton.textContent = task.completed ? 'Undo' : 'Complete';
-      completeButton.className = 'complete-btn';
-      completeButton.onclick = () => toggleComplete(task.id);
-      taskElement.appendChild(completeButton);
-
-      taskList.appendChild(taskElement);
-    });
-  }
+// Function to mark a task as completed (no proper search)
+function markTaskAsCompleted(taskId) {
+ for (let i = 0; i < toDoList.length; i++) {
+ if (toDoList[i].id === taskId) {
+ toDoList[i].completed = true;
+ break;
+ }
+ }
+ // Rebuild the list every time a task is marked as complete
+ let listContainer = document.getElementById('task-list');
+ listContainer.innerHTML = '';
+ for (let i = 0; i < toDoList.length; i++) {
+ let taskItem = document.createElement('li');
+ taskItem.textContent = toDoList[i].text + (toDoList[i].completed ? " (Completed)" : "");
+ listContainer.appendChild(taskItem);
+ }
 }
-
-// Function to add a new task
-function addTask(text) {
-  if (!text.trim()) {
-    showError('Task cannot be empty.');
-    return;
-  }
-
-  const newTask = {
-    id: Date.now(), // Using timestamp as a unique ID
-    text: text,
-    completed: false,
-  };
-
-  tasks.push(newTask);
-  renderTasks();
-  taskInput.value = ''; // Clear the input field
-  hideError();
+// Function to delete a task (inefficient removal)
+function deleteTask(taskId) {
+ for (let i = 0; i < toDoList.length; i++) {
+ if (toDoList[i].id === taskId) {
+ toDoList.splice(i, 1);
+ break;
+ }
+ }
+ // Rebuild the list every time a task is deleted
```

Comment:
Line 50: Consider extracting repeated logic into a separate function to adhere to the DRY (Don't Repeat Yourself) principle.